src/BlockTypes/SingleProduct.php

```diff
@@ -136,13 +138,13 @@ protected function replace_post_for_single_product_inner_block( $block, &$contex
 				if ( 'core/post-excerpt' === $block_name || 'core/post-title' === $block_name ) {
 					global $post;
 					// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
-					$post = get_post( $this->product_id );
-					setup_postdata( $post );
+					$post                         = get_post( $this->product_id );
+					$global_post_variable_changed = setup_postdata( $post );
 				}
 				$context['postId'] = $this->product_id;
 			}
 
-			if ( ! $this->single_product_inner_blocks_names ) {
+			if ( ! $this->single_product_inner_blocks_names && $global_post_variable_changed ) {
 				wp_reset_postdata();
```

Comment:
I'm wondering if we should revert the $global_post_variable_changed to its original state after invoking wp_reset_postdata(). This precaution may be necessary if the class instance is called again; otherwise, I believe that once we set the $global_post_variable_changed for the first time, it will consistently hold a true value and never return to a false state (unless the memory is cleaned, for instance, after a page refresh).

Reply:
> I'm wondering if we should revert the $global_post_variable_changed to its original state after invoking wp_reset_postdata().

Yeah, it doesn't hurt to change it back as an extra precaution. ✅